src/Files.Uwp/Helpers/LayoutPreferences/LayoutPrefsDb.cs

```diff
@@ -0,0 +1,143 @@
+﻿using LiteDB;
+using System;
+using System.Linq;
+using Files.Shared.Extensions;
+
+namespace Files.Uwp.Helpers.LayoutPreferences
+{
+    public class LayoutPrefsDb : IDisposable
+    {
+        private readonly LiteDatabase db;
+
+        public LayoutPrefsDb(string connection, bool shared = false)
+        {
+            db = new LiteDatabase(new ConnectionString(connection)
+            {
+                Mode = shared ? FileMode.Shared : FileMode.Exclusive
+            }, new BsonMapper() { IncludeFields = true });
+        }
+
+        public void SetPreferences(string filePath, ulong? frn, LayoutPreferences prefs)
+        {
+            // Get a collection (or create, if doesn't exist)
+            var col = db.GetCollection<LayoutDbPrefs>("layoutprefs");
+
+            var tmp = _FindPreferences(filePath, frn);
+            if (tmp == null)
+            {
+                if (prefs != null)
+                {
+                    // Insert new tagged file (Id will be auto-incremented)
+                    var newPref = new LayoutDbPrefs
+                    {
+                        FilePath = filePath,
+                        Frn = frn,
+                        Prefs = prefs
+                    };
+                    col.Insert(newPref);
```

Comment:
Should we also call `col.ensureIndex(...)` to make lookup via the Filepath potentially faster?

Reply:
Done, both on filepath and frn 👍

Reply:
I think we should ensure the index if we insert, not if we look up the entities right? Or maybe I misunderstood what it does but creating the index while searching is not ideal, creating it before you run your query is better.

Reply:
Moving it after the insert sounds right, thanks. 👍